superflore/generators/bitbake/yocto_recipe.py

```diff
@@ -51,11 +54,30 @@ def __init__(self, name, distro, src_uri, tar_dir):
         self.archive_name = None
         self.license_md5 = None
         self.tar_dir = tar_dir
-        self.downloadArchive()
-        self.src_sha256 = hashlib.sha256(
-            open(self.getArchiveName(), 'rb').read()).hexdigest()
-        self.src_md5 = hashlib.md5(
-            open(self.getArchiveName(), 'rb').read()).hexdigest()
+        if md5_cache is not None:
+            if self.getArchiveName() not in md5_cache:
+                self.downloadArchive()
```

Comment:
Ahh, this is likely the performance bottleneck. It has to download the archive before computing the hash.